src/main/java/com/onionshop/events/CanvasEvents.java

```diff
@@ -1,20 +1,98 @@
 package com.onionshop.events;
 
+import com.onionshop.entities.Colour;
 import com.onionshop.managers.DrawingManager;
+import javafx.scene.control.Button;
 import javafx.scene.input.MouseEvent;
+import javafx.scene.paint.Color;
 
 public class CanvasEvents {
     DrawingManager currentDrawingManager;
+    //This is used to convert between our colours (0-255) and javaFx's colors (0.0-1.0)
+    private final float conversionValue = 255;
 
     public CanvasEvents(DrawingManager projectDrawingManager){
         this.currentDrawingManager = projectDrawingManager;
     }
 
+    /**
+     * Processes the mouse data for the drawing manager and passes the returned updated pixels
+     * back from the drawing manager after drawing is finished on the backend
+     *
+     * @param inputMouseData the current x y coordinate of the mouse
+     * @return
+     */
     public int[][] processControllerDataForDrawingManager(MouseEvent inputMouseData) {
         int x = (int)inputMouseData.getX();
         int y = (int)inputMouseData.getY();
         int[][] pixelsToReturn = currentDrawingManager.updateCanvasAfterStroke(x, y);
         return pixelsToReturn;
     }
 
+    /**
+     * Creates our backend's version of a colour based on the JavaFx color passed in. This Colour is then sent
+     * to the drawing manager to update drawing on the backend. Returns the
+     *
+     * @param userSelectedColour The selected color from the frontend
+     * @return The color to select on the frontend
+     */
+    public Color processSelectedColour(Color userSelectedColour) {
```

Comment:
Im not sure how to do this in Java, but maybe in our imports we a can import this as JFXColour (as an alias) or something so we don't confuse it with our Colour class. Also Im glad were training the American to spell colour the correct way, lol.